src/index.ts

```diff
@@ -324,6 +324,35 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
 
           const comments = model.comments;
           let index = comments.length;
+          let { start, end } = editorWidget.editor.getSelection();
+          //backwards selection compatibility
+          if (
+            start.line > end.line ||
+            (start.line === end.line && start.column > end.column)
+          ) {
+            [start, end] = [end, start];
+          }
+
+          for (let i = comments.length; i > 0; i--) {
+            const comment = comments.get(i - 1) as ITextSelectionComment;
+            let sel = comment.target;
+            let commentStart = sel.start;
+            //backwards selection compatibility
+            if (
```

Comment:
Would a selection ever be backwards if the comment factory for text selection comments makes sure they go in in the right order?

Reply:
Looks like the comment factory doesn't input them in the right order for backwards selections; I'll get onto that